services/core-api/app/api/now_applications/models/now_application.py

```diff
@@ -139,11 +187,17 @@ def __repr__(self):
 
     @classmethod
     def find_by_application_id(cls, now_application_id):
-        return cls.query.filter_by(now_application_id=now_application_id).first()
+        return cls.query.filter_by(
+            now_application_id=now_application_id).first()
 
     @classmethod
     def validate_guid(cls, guid, msg='Invalid guid.'):
         try:
             uuid.UUID(str(guid), version=4)
         except ValueError:
             raise AssertionError(msg)
+
+    def save(self, commit=True):
```

Comment:
Is this function necessary? I think the `onupdate=...` you defined in the model should do this for you already.